pkg/v1/cmd.go

```diff
@@ -178,7 +178,12 @@ func Run(ctx context.Context, logger *logrus.Logger, opts Options) error {
 				return fmt.Errorf("Failed to push changes.: %w", err)
 			}
 
-			var labelsToAdd []string
+			labelsToAdd := []string{
+				// The repos is set to use rebase merge method for making it easier to programmatically
+				// determine the commits which need to be carried. But the sync PR itself need to use merge.
+				// By adding this label we instruct tide to merge instead of using the default behaviour.
```

Comment:
My understanding of why we might need to use the "merge" method for the sync PRs is related to how the commits are synced and the drop/carry commits are handled. I'm not certain there would be an issue, but I think in general we are trying to keep the sync process the same but update the merge process for anything that is not a sync to use the "rebase" method to ensure drop/carry commits are merged on top of the most recent sync and are appropriately accounted for.

Reply:
could we go a bit deeper here? It would be good if we could point to some documentation on how the sync process works and maybe where the decision to do sync PR one way and everything else another way is documented?

Reply:
@perdasilva I _think_ the way the sync process works is that it attempts to "rewrite" history with the drop/carry commits. I don't know for certain that this would cause an issue if the merge method was "rebase", but I think it makes the most sense to change one thing at a time to fix our current issue. 

In this case, adding the label ensures that the one change we are making is that all non-sync PRs (which should be some form of drop/carry commit) are merged via the "rebase" merge method to ensure they are merged on top of the most recent changes (ensuring they get merged on top of the most recent sync and are appropriately tracked during the sync process).

This document should outline the sync process in more detail: https://docs.google.com/document/d/1DrLUbo0CByTDv9Jxc5RJCqZgG5Ivlc-BCh_qASQooq0/edit#heading=h.cwvtj4ajzja1